src/amo/components/UserProfile/index.js

```diff
@@ -187,8 +222,18 @@ export function mapStateToProps(
   state: {| users: UsersStateType |},
   ownProps: Props,
 ) {
+  const currentUser = getCurrentUser(state.users);
+  const user = getUserByUsername(state.users, ownProps.params.username);
+
+  let hasEditPermission = currentUser && user && currentUser.id === user.id;
+  if (currentUser && hasPermission(state, USERS_EDIT)) {
```

Comment:
You could have done this with an `||` above, but I guess you wanted to avoid a complicated conditional statement?